latch_cli/snakemake/workflow.py

```diff
@@ -0,0 +1,490 @@
+
+    def compile(self, **kwargs):
+
+        self._input_parameters = interface_to_parameters(self.python_interface)
+
+        GLOBAL_START_NODE = Node(
+            id=_common_constants.GLOBAL_INPUT_NODE_ID,
+            metadata=None,
+            bindings=[],
+            upstream_nodes=[],
+            flyte_entity=None,
+        )
+
+        node_map = {}
+
+        target_files = [x for job in self._dag.targetjobs for x in job.input]
+
+        for layer in self._dag.toposorted():
+            for job in layer:
+
+                is_target = False
+
+                if job in self._dag.targetjobs:
+                    continue
+
+                target_file_for_output_param: Dict[str, str] = {}
+                target_file_for_input_param: Dict[str, str] = {}
+
+                python_outputs: Dict[str, Type] = {}
+                for x in job.output:
+                    if x in target_files:
+                        is_target = True
+                    param = variable_name_for_value(x, job.output)
+                    target_file_for_output_param[param] = x
+                    python_outputs[param] = LatchFile
+
+                @dataclass
+                class JobOutputInfo:
+                    jobid: str
+                    output_param_name: str
+
+                dep_outputs = {}
+                for dep, dep_files in self._dag.dependencies[job].items():
+                    for o in dep.output:
+                        if o in dep_files:
+                            dep_outputs[o] = JobOutputInfo(
+                                jobid=dep.jobid,
+                                output_param_name=variable_name_for_value(
+                                    o, dep.output
+                                ),
+                            )
+
+                python_inputs: Dict[str, Type] = {}
+                promise_map: Dict[str, str] = {}
+                for x in job.input:
+                    param = variable_name_for_value(x, job.input)
+                    target_file_for_input_param[param] = x
+                    python_inputs[param] = LatchFile
+                    if x in dep_outputs:
+                        promise_map[param] = dep_outputs[x]
+
+                interface = Interface(python_inputs, python_outputs, docstring=None)
+                task = SnakemakeJobTask(
+                    job=job,
+                    inputs=python_inputs,
+                    outputs=python_outputs,
+                    target_file_for_input_param=target_file_for_input_param,
+                    target_file_for_output_param=target_file_for_output_param,
+                    is_target=is_target,
+                    interface=interface,
+                )
+                self.snakemake_tasks.append(task)
+
+                typed_interface = transform_interface_to_typed_interface(interface)
+                self.interface.inputs.keys()
+                bindings = []
+                for k in sorted(interface.inputs):
```

Comment:
why sorted? what is the sort by?

Reply:
borrowed from flyte, not sure, will take it out and see what/if something breaks